core/deployment/src/main/java/io/quarkus/deployment/dev/testing/TestSupport.java

```diff
@@ -0,0 +1,258 @@
+        this.curatedApplication = curatedApplication;
+        this.compilationProviders = compilationProviders;
+        this.context = context;
+    }
+
+    public static Optional<TestSupport> instance() {
+        if (RuntimeUpdatesProcessor.INSTANCE == null) {
+            return Optional.empty();
+        }
+        return Optional.ofNullable(RuntimeUpdatesProcessor.INSTANCE.getTestSupport());
+    }
+
+    public boolean isRunning() {
+        if (testRunner == null) {
+            return false;
+        }
+        return testRunner.isRunning();
+    }
+
+    public List<TestListener> getTestListeners() {
+        return testListeners;
+    }
+
+    /**
+     * returns the current status of the test runner.
+     * <p>
+     * This is expressed in terms of test run ids, where -1 signifies
+     * no result.
+     */
+    public RunStatus getStatus() {
+        if (testRunner == null) {
+            return new RunStatus(-1, -1);
+        }
+        long last = -1;
+        //get the running test id before the current status
+        //otherwise there is a race where they both could be -1 even though it has started
+        long runningTestRunId = testRunner.getRunningTestRunId();
+        TestRunResults tr = testRunResults;
+        if (tr != null) {
+            last = tr.getId();
+        }
+        return new RunStatus(last, runningTestRunId);
+    }
+
+    public void start() {
+        if (!started) {
+            synchronized (this) {
+                if (!started) {
+                    try {
+                        if (context.getApplicationRoot().getTest().isPresent()) {
+                            started = true;
+                            init();
+                            for (TestListener i : testListeners) {
+                                i.testsEnabled();
+                            }
+                            testRunner.enable();
+                        }
+                    } catch (Exception e) {
+                        log.error("Failed to create compiler, runtime compilation will be unavailable", e);
+                    }
+
+                }
+            }
+        }
+    }
+
+    public void init() {
+        if (!context.getApplicationRoot().getTest().isPresent()) {
+            return;
+        }
+        if (testCuratedApplication == null) {
+            try {
+                testCuratedApplication = curatedApplication.getQuarkusBootstrap().clonedBuilder()
+                        .setMode(QuarkusBootstrap.Mode.TEST)
+                        .setDisableClasspathCache(true)
+                        .setIsolateDeployment(true)
+                        .setTest(true)
+                        .setAuxiliaryApplication(true)
+                        .addAdditionalApplicationArchive(new AdditionalDependency(
+                                Paths.get(context.getApplicationRoot().getTest().get().getClassesPath()), true,
+                                true))
+                        .build()
```

Comment:
Not necessarily for this PR, I am wondering whether we could pass the `AppModel` instance to the test app instead of re-resolving it. That could save some time.

Reply:
This is a TEST appmodel, but the outer one is DEV. I want to talk to you about this though, there is a lot of stuff that could be cleaned up here.

Reply:
Right, makes sense.